# Langkah 1 - Load Data

In [4]:
import pandas as pd # Melakukan Import Library Pandas

dpath = 'data/Titanic-Dataset.csv' # Menentukan Path Dataset Titanic
df = pd.read_csv(dpath) # Membaca Dataset Titanic dan Menyimpannya ke dalam DataFrame Pandas
df.head() # Menampilkan 5 Data Teratas karena tidak ada argumen di dalam fungsi head(), jika fungsi head() terdapat argumen berupa angka maka akan menampilkan data sebanyak angka yang ada dalam argumen

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

# Langkah 2 - Pengecekan Data

In [5]:
df.info() # Menampilkan Informasi Data Titanic yang ada dalam DataFrame Pandas, informasi tersebut berisi seperti jumlah data setiap kolom, tipe data setiap kolom, jumlah kolom, dan jumlah baris yang ada dalam dataset.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [6]:
df.isnull().sum() # Menampilkan jumlah data yang kosong atau hilang pada setiap kolom

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Berdasarkan hasil diatas, ditemukan bahwa terdapat 177 data yang hilang pada kolom Age, 687 data yang hilang pada kolom Cabin, dan 2 data yang hilang pada kolom Embarked.

# Langkah 3 - Imputasi

In [8]:
# Age - mean
df['Age'].fillna(value=df['Age'].mean(), inplace=True) # Mengisi data yang kosong atau hilang pada kolom Age dengan nilai rata-rata dari kolom Age. Hal ini dikarenakan kolom Age adalah data nominal yang ditunjukkan oleh tipe data float64, sehingga imputasi yang tepat adalah menggunakan rata-rata/mean.

# Cabin - "DECK"
df['Cabin'].fillna(value="DECK", inplace=True) # Mengisi data yang kosong atau hilang pada kolom Cabin dengan nilai "DECK". Hal ini dikarenakan kita berasumsi bahwa penumpang yang tidak memiliki nomor cabin tinggal di dek kapal.

# Embarked - modus
df['Embarked'].fillna(value=df['Embarked'].mode, inplace=True) # Mengisi data yang kosong atau hilang pada kolom Embarked dengan nilai modus dari kolom Embarked. Hal ini dikarenakan kolom Embarked adalah data nominal berdasarkan nilai pada kolom Embarked, sehingga imputasi yang tepat adalah menggunakan modus.

# Langkah 4 - Validasi Hasil

In [9]:
df.info() # Menampilkan informasi data untuk melakukan validasi hasil atas proses imputation yang telah dilakukan.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        891 non-null    object 
 11  Embarked     891 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [10]:
df.isnull().sum() # Kemudian dilakukan validasi lagi untuk memastikan sudah tidak ada data yang kosong.

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Cabin          0
Embarked       0
dtype: int64

In [11]:
df.head(10) # Menampilkan 10 data teratas untuk memastikan hasil imputasi sudah sesuai dengan yang diharapkan.

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex        Age  \
0                            Braund, Mr. Owen Harris    male  22.000000   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.000000   
2                             Heikkinen, Miss. Laina  female  26.000000   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.000000   
4                           Allen, Mr. William Henry    male  35.000000   
5                                   Moran, Mr. James    male  29.699118   
6                            McCarthy, Mr. Timothy J    male  54.000000   
7                     Palsson, Master. Gosta Leonard    male   2.000000   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.000000   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.000000   

   SibSp  Parch            Ticket     Fare Cabin Embarked  
0      1      0         A/5 21171   7.2500  DECK        S  
1      1      0          PC 17599  71.2833   C85        C  
2      0      0  STON/O2. 3101282   7.9250  DECK        S  
3      1      0            113803  53.1000  C123        S  
4      0      0            373450   8.0500  DECK        S  
5      0      0            330877   8.4583  DECK        Q  
6      0      0             17463  51.8625   E46        S  
7      3      1            349909  21.0750  DECK        S  
8      0      2            347742  11.1333  DECK        S  
9      1      0            237736  30.0708  DECK        C